scripts/validate_staging_repo.sh

```diff
@@ -0,0 +1,133 @@
+plugins {
+    id 'application'
+}
+
+repositories {
+    mavenCentral()
+    maven {
+        url '${STAGING_REPO}'
+    }
+}
+
+
+dependencies {
+    implementation "org.apache.pulsar:pulsar-client-reactive-adapter:${VERSION}"
+    implementation "org.apache.pulsar:pulsar-client-reactive-producer-cache-caffeine:${VERSION}"
+}
+
+application {
+    mainClass = 'HelloPulsarClientReactive'
+}
+EOF
+
+cat >settings.gradle <<EOF
+rootProject.name = 'test-app-reactive'
+EOF
+
+mkdir -p src/main/java
+cat >src/main/java/HelloPulsarClientReactive.java <<EOF
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import org.apache.pulsar.client.api.PulsarClient;
+import org.apache.pulsar.client.api.PulsarClientException;
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.reactive.client.adapter.AdaptedReactivePulsarClientFactory;
+import org.apache.pulsar.reactive.client.api.MessageSpec;
+import org.apache.pulsar.reactive.client.api.ReactiveMessagePipeline;
+import org.apache.pulsar.reactive.client.api.ReactiveMessageSender;
+import org.apache.pulsar.reactive.client.api.ReactivePulsarClient;
+import reactor.core.publisher.Mono;
+
+public class HelloPulsarClientReactive {
+
+    public static void main(String[] args) throws PulsarClientException, InterruptedException {
+        // Before running this, start Pulsar within docker with this command:
+        // docker run -it -p 8080:8080 -p 6650:6650 apachepulsar/pulsar:3.1.1 /pulsar/bin/pulsar standalone -nss -nfw
+
+        try (PulsarClient pulsarClient = PulsarClient.builder().serviceUrl("pulsar://localhost:6650").build()) {
+
+            ReactivePulsarClient reactivePulsarClient = AdaptedReactivePulsarClientFactory.create(pulsarClient);
+
+            CountDownLatch latch = new CountDownLatch(3);
+
+            ReactiveMessagePipeline reactiveMessagePipeline = reactivePulsarClient.messageConsumer(Schema.STRING)
+                    .subscriptionName("sub").topic("testreactive").build().messagePipeline()
+                    .messageHandler(message -> Mono.fromRunnable(() -> {
+                        System.out.println("Received message:" + message.getValue());
+                        latch.countDown();
+                    })).build().start();
+
+            ReactiveMessageSender<String> messageSender = reactivePulsarClient.messageSender(Schema.STRING)
+                    .topic("testreactive").cache(AdaptedReactivePulsarClientFactory.createCache()).build();
+
+            messageSender.sendOne(MessageSpec.of("Hello world!")).block();
+            messageSender.sendOne(MessageSpec.of("Hello Pulsar!")).block();
+            messageSender.sendOne(MessageSpec.of("Hello Pulsar Client Reactive!")).block();
+
+            if(!latch.await(10, TimeUnit.SECONDS)) {
+                throw new RuntimeException("Did not receive all messages");
+            }
+            reactiveMessagePipeline.stop();
+        }
+    }
+
+}
+EOF
+
+gradle build
```

Comment:
We have gradlew in the repo so we should use it

Reply:
this script isn't run within the repository. gradlew isn't present in this case.